### PR TITLE
[CI] Test OOT credential provider

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
[CI] Test OOT credential provider
* Add a template
* Add an e2e test
#### Which issue(s) this PR fixes:

Fixes #
Related https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2403

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
where is the kubelet setting code for credential provider?

**Comment by andyzhangx** [comment]
> where is the kubelet setting code for credential provider?

nvm, it's here: [manifest/cluster-api/cluster-template-prow-ci-version-oot-credential-provider.yaml](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/4469/files#diff-ed8e8a4af4bff2f7bcdccbd1c66b02b49ced891b8e635d0e7ae4895cbc8ba32b)

File: tests/e2e/auth/cred.go [hunk 1]
```diff
@@ -88,22 +90,63 @@ var _ = Describe("Azure Credential Provider", Label(utils.TestSuiteLabelCredenti
 		err = utils.DockerLogout()
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	It("should be able to create an ACR cache and pull images from it", Label(utils.TestSuiteLabelOOTCredential), func() {
```
Comment (andyzhangx) [inline tests/e2e/auth/cred.go:94]: what do you mean `create an ACR cache`?

Comment (lzhecheng) [reply tests/e2e/auth/cred.go:94]: I am using a preview feature of ACR: [acr cache](https://learn.microsoft.com/en-us/azure/container-registry/tutorial-registry-cache)
The reason is that I want to test Windows but control-plane Node is Linux. So, it is hard to push a Windows image to ACR. With this new feature, I can skip the pushing.

Comment (andyzhangx) [reply tests/e2e/auth/cred.go:94]: thanks, pls add this comment into the test suites, that would make it more clear.

Comment (lzhecheng) [reply tests/e2e/auth/cred.go:94]: Makes sense. Updated.

**Pull request merged**